nvtabular/io.py

```diff
@@ -228,55 +235,68 @@ def __init__(
         # Resolve file system
         self.fs = fs or get_fs_token_paths(str(out_dir))[0]
 
-        # create thread queue and locks
-        self.queue = queue.Queue(num_threads)
-        self.write_locks = [threading.Lock() for _ in range(num_out_files)]
+        # Only use threading if num_threads > 0
+        if self.num_threads:
 
-        # signifies that end-of-data and that the thread should shut down
-        self._eod = object()
+            # create thread queue and locks
+            self.queue = queue.Queue(num_threads)
```

Comment:
nitpick: we should set the queue to None when not in use (rather than just not have on the object at all)